### DISCUSSION TITLE
Logging From Builtin Function

### DISCUSSION THREAD

**Discussion by abdullahalrifat** [open-policy-agent/community#326] [Community]
I have created a custom built in function following this [adding-built-in](https://www.openpolicyagent.org/docs/latest/extensions/#adding-built-in-functions-to-the-opa-runtime), and create a OPA runtime. Now I want to log any error happening in the built in function written in go to come as log inside OPA container.
can anyone help me how to do that?

**Comment by charlieegan3** [comment]
Hi Abdullah, might the BuiltinContext PrintHook be suitable for your needs?

```
	rego.RegisterBuiltin2(
		&rego.Function{
			Name:    "github.repo",
			Decl:    types.NewFunction(types.Args(types.S, types.S), types.A),
			Memoize: true,
		},
		func(bctx rego.BuiltinContext, a, b *ast.Term) (*ast.Term, error) {
			err := bctx.PrintHook.Print(print.Context{}, "github.repo called")
			if err != nil {
				return nil, err
			}
...

```
